provider-announcements/src/java/org/sakaiproject/archiver/provider/AnnouncementsArchiver.java

```diff
@@ -0,0 +1,117 @@
+package org.sakaiproject.archiver.provider;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.sakaiproject.announcement.api.AnnouncementMessage;
+import org.sakaiproject.announcement.api.AnnouncementService;
+import org.sakaiproject.archiver.api.Archiveable;
+import org.sakaiproject.archiver.api.ArchiverRegistry;
+import org.sakaiproject.archiver.api.ArchiverService;
+import org.sakaiproject.archiver.util.Jsonifier;
+import org.sakaiproject.content.api.ContentHostingService;
+import org.sakaiproject.entity.api.Reference;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.exception.PermissionException;
+import org.sakaiproject.exception.ServerOverloadException;
+import org.sakaiproject.exception.TypeException;
+import org.sakaiproject.message.api.Message;
+import org.sakaiproject.site.api.SiteService;
+
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Implementation of {@link Archiveable} for the announcements tool
+ */
+
+@Slf4j
+public class AnnouncementsArchiver implements Archiveable {
+
+	private static final String ANNOUNCEMENTS_TOOL = "sakai.announcements";
+
+	public void init() {
+		ArchiverRegistry.getInstance().register(ANNOUNCEMENTS_TOOL, this);
+	}
+
+	public void destroy() {
+		ArchiverRegistry.getInstance().unregister(ANNOUNCEMENTS_TOOL);
+	}
+
+	@Setter
+	private ArchiverService archiverService;
+	
+	@Setter 
+	private AnnouncementService announcementService;
+	
+	@Setter
+	private ContentHostingService contentHostingService;
+
+	@Override
+	public void archive(final String archiveId, final String siteId, final String toolId, final boolean includeStudentContent) {
+		
+		try {
+			
+			// Get the announcements for this site
+			String channelRef = this.announcementService.channelReference(siteId, SiteService.MAIN_CONTAINER);
+			List<Message> announcements = new ArrayList<Message>();
```

Comment:
If you made this AnnoucnementMessafe then you wouldn't need the cast in the for loop. Also these three lines could be in a separate method to make this more readable.